src/Learning/SFA.DAS.Learning/Application/Notification/ExtendedNotificationService.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Microsoft.Extensions.Logging;
+using SFA.DAS.Encoding;
+using SFA.DAS.Learning.InnerApi;
+using SFA.DAS.Notifications.Messages.Commands;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Extensions;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests.Commitments;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests.Learning;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses.Commitments;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses.Learning;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.Learning.Application.Notification;
+
+public interface IExtendedNotificationService
+{
+    Task<CommitmentsApprenticeshipDetails> GetApprenticeship(GetCurrentPartyIdsResponse currentPartyIds);
+    Task<GetCurrentPartyIdsResponse> GetCurrentPartyIds(Guid apprenticeshipKey);
+    Task<IEnumerable<Recipient>> GetEmployerRecipients(long accountId);
+    Task<IEnumerable<Recipient>> GetProviderRecipients(long providerId);
+    Task<bool> Send(Recipient recipient, string templateId, Dictionary<string, string> tokens);
+}
+
+public class ExtendedNotificationService : IExtendedNotificationService
+{
+    protected ILogger<ExtendedNotificationService> _logger;
+    private readonly IAccountsApiClient<AccountsConfiguration> _accountsApiClient;
+    private readonly ILearningApiClient<LearningApiConfiguration> _learningApiClient;
+    private readonly ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> _apiCommitmentsClient;
+    private readonly IProviderAccountApiClient<ProviderAccountApiConfiguration> _providerAccountApiClient;
+    private readonly IEncodingService _encodingService;
+    private readonly INotificationService _notificationService;
+
+    public ExtendedNotificationService(
+        ILogger<ExtendedNotificationService> logger,
+        IAccountsApiClient<AccountsConfiguration> accountsApiClient,
+        ILearningApiClient<LearningApiConfiguration> learningApiClient,
+        ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> apiCommitmentsClient,
+        IEncodingService encodingService,
+        INotificationService notificationService,
+        IProviderAccountApiClient<ProviderAccountApiConfiguration> providerAccountApiClient)
+    {
+        _logger = logger;
+        _accountsApiClient = accountsApiClient;
+        _learningApiClient = learningApiClient;
+        _apiCommitmentsClient = apiCommitmentsClient;
+        _encodingService = encodingService;
+        _notificationService = notificationService;
+        _providerAccountApiClient = providerAccountApiClient;
+    }
+
+    /// <summary>
+    /// Call to apprenticeship inner to get the apprenticeships current provider and employer ids
```

Comment:
learning inner? Unsure about apprenticeshipKey here though as we are in notifications domain which is scoped to apprenticeships specifically. I have not commented on similar places where apprenticeships are referenced as part of either notifications or directly from commitments as I think you have deliberately and correctly left them as apprenticeships.